numpyro/infer/hmc.py

```diff
@@ -608,9 +602,9 @@ def __init__(
         potential_fn=None,
         kinetic_fn=None,
         step_size=1.0,
+        num_steps=None,
```

Comment:
I think we need to move this to near `trajectory_length` argument and force keyword arguments for it to make it less likely to break the current users' code: `... target_accept_prob=0.8, *, num_steps=None, trajectory_length=2 * pi`. Could you also add docstring for it like in `hmc`?